govcd/api_vcd_versions.go

```diff
@@ -231,13 +231,23 @@ func (client *Client) validateAPIVersion() error {
 	return nil
 }
 
-// GetSpecificApiVersionOnCondition returns default version or wantedApiVersion if it is connected to version
-// described in vcdApiVersionCondition
-// f.e. values ">= 32.0", "32.0" returns 32.0 if vCD version is above or 9.7
-func (client *Client) GetSpecificApiVersionOnCondition(vcdApiVersionCondition, wantedApiVersion string) string {
+// GetSpecificApiVersionOnCondition returns default version or wantedApiVersion if it is connected
+// to version described in vcdApiVersionCondition e.g. values ">= 32.0", "32.0" returns 32.0 if vCD
+// version is above or 9.7
+// Note. This function will always respect minimum supported API version which is defined in
+// client.APIVersion. If the wantedApiVersionOrMinimumRequired is less than minimum supported
+// version, this function will return the minimum supported version
+func (client *Client) GetSpecificApiVersionOnCondition(vcdApiVersionCondition, wantedApiVersionOrMinimumRequired string) string {
```

Comment:
Would be useful to have a paragraph explaining the risk involved here (connecting to a newer API version than originally developed against).

Reply:
Added